bin/crew

```diff
@@ -1506,6 +1524,25 @@ def upload(pkg_name = nil, pkg_version = nil, gitlab_token = nil, binary_compres
         end
       end
     end
+    # Upload python wheels if we are dealing with a pip package, but only
+    # if a gitlab token username is set. (The generic repo does not
+    # require a gitlab token username.)
+    next unless Kernel.system "grep -q \"^require 'buildsystems/pip'\" #{pkg_file}"
+    if gitlab_token_username.nil?
+      puts "\nGITLAB_TOKEN_USERNAME environment variable not set, so generated python wheels will not be uploaded.\n".lightred
+      next package
```

Comment:
Should we abort here instead of next to avoid a stampede of the same messages?

Reply:
No? Because then at least builds will continue?

Generated python wheels can always be uploaded later....

Reply:
What does uploading later really mean?  My thought was if `GITLAB_TOKEN_USERNAME` is not set, it should fail early so the wheels are included?

Reply:
The wheels get saved in the container in `/usr/local/.cache/pip`.

As long as the container isn't closed, the wheels are still there. But at this point yes, it probably makes sense to just set an abort... It's easy enough to make sure everyone generates a GITLAB_TOKEN_USERNAME...